lib/rubocop/cop/naming/constant_name.rb

```diff
@@ -28,7 +28,8 @@ def on_casgn(node)
           # NewClass = something_that_returns_a_class
           # It's also ok to assign a class constant another class constant
           # SomeClass = SomeOtherClass
-          return if value && %i[send block const casgn].include?(value.type)
+          return if value && (%i[block const casgn].include?(value.type) ||
+                    value.send_type? && value.method_name != :freeze)
```

Comment:
I think this can be extended to a whole list of safe methods - `to_s`, `to_a`, etc.

Reply:
Thanks for the review.
I see. For example, I think that we can prepare the following settings in default.yml.

```yaml
Naming/ConstantName
  Whitelist:
    - freeze
```

However, I don't know the following things.

- How much false negatives occur in cases other than `freeze`
- Criteria for preparing defaults other than `to_s`,` to_a`

How about opening another PR that sets `freeze` to `Whitelist` in default.yml?
Or is there any other good idea?

Reply:
Off the top of my head, shouldn't it rather check that first receiver is a literal?

Reply:
Maybe it should not be a whitelist, be a blacklist.

For example:

```ruby
def_node_matcher :class_return_method?, <<-P
  {
    (send (const _ :Class) :new ...)
    ...
  }
P
return if value && (%i[block const casgn].include?(value.type) || class_return_method?(value))
```

And I think RuboCop should give up on `some_user_defined_method`, or we can add `ReturnClassMethods`.

What do you think?

Reply:
> Maybe it should not be a whitelist, be a blacklist.

I don't think this will work, because we don't know what methods return at runtime, so marking all other method sends as offenses is probably too aggressive. It is not on par with how we handle method sends in other cops.

> And I think RuboCop should give up on some_user_defined_method, or we can add ReturnClassMethods.
>
> What do you think?

I am also inclined to think that this cop doesn't quite need a configurable whitelist.

Reply:
Thank you for the ideas. I will try to improve by using only AST without using configurable whitelist or blacklist.

Reply:
I fixed that following your advices. 688acd1